src/main/java/org/lwjglx/opengl/Display.java

```diff
@@ -642,19 +646,21 @@ public static int setIcon(java.nio.ByteBuffer[] icons) {
             return 0;
         }
         GLFWImage.Buffer glfwImages = GLFWImage.calloc(icons.length);
-        ByteBuffer[] nativeBuffers = new ByteBuffer[icons.length];
-        for (int icon = 0; icon < icons.length; icon++) {
-            nativeBuffers[icon] = org.lwjgl.BufferUtils.createByteBuffer(icons[icon].capacity());
-            nativeBuffers[icon].put(icons[icon]);
-            nativeBuffers[icon].flip();
-            int dimension = (int) Math.sqrt(nativeBuffers[icon].limit() / 4D);
-            if (dimension * dimension * 4 != nativeBuffers[icon].limit()) {
-                throw new IllegalStateException();
+        try (MemoryStack stack = stackPush()) {
```

Comment:
The scope of this must include `GLFW.glfwSetWindowIcon(getWindow(), glfwImages);`, otherwise that function gets called with already freed memory